server/lib/tuist_web/controllers/api/oidc_controller.ex

```diff
@@ -0,0 +1,189 @@
+defmodule TuistWeb.API.OIDCController do
+  @moduledoc """
+  Controller for OIDC token exchange.
+
+  This controller handles the exchange of CI provider OIDC tokens for
+  short-lived Tuist access tokens. Currently supports GitHub Actions.
+  """
+
+  use OpenApiSpex.ControllerSpecs
+  use TuistWeb, :controller
+
+  alias OpenApiSpex.Schema
+  alias Tuist.Guardian
+  alias Tuist.OIDC
+  alias Tuist.Projects
+  alias TuistWeb.API.Schemas.Error
+
+  @providers %{
+    "https://token.actions.githubusercontent.com" => %{
+      jwks_uri: "https://token.actions.githubusercontent.com/.well-known/jwks",
+      repository_claim: "repository"
+    }
+    # Future providers:
+    # "https://gitlab.com" => %{
+    #   jwks_uri: "https://gitlab.com/oauth/discovery/keys",
+    #   repository_claim: "project_path"
+    # }
+  }
+
+  plug(
+    OpenApiSpex.Plug.CastAndValidate,
+    json_render_error_v2: true,
+    render_error: TuistWeb.RenderAPIErrorPlug
+  )
+
+  tags(["OIDC Authentication"])
+
+  @default_ci_scopes [
```

Comment:
in the future, we can consider making the scopes configurable, but for now, I'd keep it simple and include all the scopes we think make sense to be used from the CI (similar to project tokens that OIDC tokens should mostly replace)